mlrun/api/api/endpoints/workflows.py

```diff
@@ -0,0 +1,295 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import copy
+import traceback
+import typing
+from http import HTTPStatus
+from typing import Dict
+
+import fastapi
+from fastapi.concurrency import run_in_threadpool
+from sqlalchemy.orm import Session
+
+import mlrun
+import mlrun.api.api.deps
+import mlrun.api.api.utils
+import mlrun.api.crud
+import mlrun.api.schemas
+import mlrun.api.utils.auth.verifier
+import mlrun.api.utils.clients.chief
+import mlrun.api.utils.singletons.db
+import mlrun.api.utils.singletons.project_member
+import mlrun.projects.pipelines
+from mlrun.api.api.utils import log_and_raise
+from mlrun.utils.helpers import logger
+
+router = fastapi.APIRouter()
+
+
+def _get_workflow_by_name(
+    project: mlrun.api.schemas.Project, name: str
+) -> typing.Optional[Dict]:
+    """
+    Getting workflow from project
+
+    :param project:     MLRun project
+    :param name:        workflow name
+
+    :return: workflow as a dict if project has the workflow, otherwise None
+    """
+    for workflow in project.spec.workflows:
+        if workflow["name"] == name:
+            return workflow
+    return None
+
+
+def _fill_workflow_missing_fields_from_project(
+    project: mlrun.api.schemas.Project,
+    name: str,
+    spec: mlrun.api.schemas.WorkflowSpec,
+    arguments: typing.Dict,
+) -> mlrun.api.schemas.WorkflowSpec:
+    """
+    Fill the workflow spec details from the project object, with favour to spec
+
+    :param project:     MLRun project that contains the workflow
+    :param name:        workflow name
+    :param spec:        workflow spec input
+    :param arguments:   arguments to workflow
+
+    :return: completed workflow spec
+    """
+    # Verifying workflow exists in project:
+    workflow = _get_workflow_by_name(project, name)
+    if not workflow:
+        log_and_raise(
+            reason=f"workflow {name} not found in project",
+        )
+    if spec:
+        # Merge between the workflow spec provided in the request with existing
+        # workflow while the provided workflow takes precedence over the existing workflow params
+        workflow = copy.deepcopy(workflow)
+        workflow = {key: val or workflow.get(key) for key, val in spec.dict().items()}
```

Comment:
`spec` will contain always all the fields in workflow spec (because it is passed as a schema), but some of them may be `None` because the user did not pass them. So in this way we collect all possible values without override original workflow values with `None`. In the way you suggested, the `None` values can override the original values. 

Either way, I think a combination would be the best solution:
```
for key, val in spec.dict().items():
    if val:
        workflow[key] = val
```